config/ingest.go

```diff
@@ -76,6 +76,29 @@ type Ingest struct {
 	SyncTimeout Duration
 }
 
+// FileStore configures a particular file store implementation.
+type FileStore struct {
+	// Type of file store to use: "", "local", "s3"
+	Type string
+	// Configuration for storing files in local filesystem.
+	Local LocalFileStore
+	// Configuration for storing files in S3.
+	S3 S3FileStore
+}
+
+type LocalFileStore struct {
+	// Path to filesystem directory where files are stored.
+	BasePath string
+}
+
+type S3FileStore struct {
+	BasePath   string
+	BucketName string
+	Region     string
+	AccessKey  string
+	SecretKey  string
```

Comment:
`Region` will typically be set via environment variables that are well known to the AWS sdk, standardised across all languages.

`AccessKey` and `Secret(Access)Key` are typically not specified in code at all for security reasons. Instead, the authentication is done by assuming a role, for which we do not need to write any code at all. This is handled by the authentication chain standardised and built into AWS SDK. We will configure the pod in k8s-level to assume a role that would then allows it to implicitly access the configured bucket.

Following that train of thought, for S3 we then only need a bucket name, and "Key",i.e. the path to which the data is uploaded, which i suspect is the same as `BasePath`? I recommend using terminology that is consistent with that of S3, i.e. Key instead of base path. [Here](https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/gov2/s3/actions/bucket_basics.go#L103) are some examples.

Reply:
My thinking was to allow the _option_ of specifying these in indexer config instead of setting it elsewhere in the infrastructure. So, generally, a `EnvProvider` or `SharedCredentialsProvider` or `EC2RoleProvider` will be used in the credentials chain, but shouldn't there be an option to have a `StaticProvider`?

Reply:
I don't think it's needed considering the built in auth chain in vanilla AWS sdk.